From 752ba7f71c57d1816413cd0716fbfe4bc73c47d5 Mon Sep 17 00:00:00 2001
From: Daniel Borkmann <dborkman@redhat.com>
Date: Fri, 14 Nov 2014 12:38:46 -0500
Subject: [net] sctp: add transport state in /proc/net/sctp/remaddr

Message-id: <44a7d30881368f2ada0e1cc65141e29fca8f452e.1415968455.git.dborkman@redhat.com>
Patchwork-id: 100050
O-Subject: [RHEL7 PATCH net 1/2] sctp: add transport state in /proc/net/sctp/remaddr
Bugzilla: 1164214
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Marcelo Leitner <mleitner@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1164214
Upstream Status: 891310d53dcf396d6a92987da265a5ffe6c09d71 (net-next)
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=8246153
Tested: verified on beaker

Upstream Commit:
commit 891310d53dcf396d6a92987da265a5ffe6c09d71
Author: Michele Baldessari <michele@acksyn.org>
Date:   Thu Oct 30 10:29:15 2014 +0100

    sctp: add transport state in /proc/net/sctp/remaddr

    It is often quite helpful to be able to know the state of a transport
    outside of the application itself (for troubleshooting purposes or for
    monitoring purposes). Add it under /proc/net/sctp/remaddr.

    Signed-off-by: Michele Baldessari <michele@acksyn.org>
    Acked-by: Neil Horman <nhorman@tuxdriver.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Daniel Borkmann <dborkman@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/sctp/proc.c b/net/sctp/proc.c
index e7d29ab..790352f 100644
--- a/net/sctp/proc.c
+++ b/net/sctp/proc.c
@@ -418,7 +418,7 @@ static void *sctp_remaddr_seq_start(struct seq_file *seq, loff_t *pos)
 
  if (*pos == 0)
   seq_printf(seq, "ADDR ASSOC_ID HB_ACT RTO MAX_PATH_RTX "
-    "REM_ADDR_RTX  START\n");
+    "REM_ADDR_RTX START STATE\n");
 
  return (void *)pos;
 }
@@ -498,7 +498,13 @@ static int sctp_remaddr_seq_show(struct seq_file *seq, void *v)
     * currently implemented, but we can record it with a
     * jiffies marker in a subsequent patch
     */
-   seq_printf(seq, "0");
+   seq_printf(seq, "0 ");
+
+   /*
+    * The current state of this destination. I.e.
+    * SCTP_ACTIVE, SCTP_INACTIVE, ...
+    */
+   seq_printf(seq, "%d", tsp->state);
 
    seq_printf(seq, "\n");
   }
-- 
1.7.1